packages/smart-wallet/src/offerWatcher.js

```diff
@@ -0,0 +1,267 @@
+import { prepareExoClassKit, watchPromise } from '@agoric/vat-data';
+import { M } from '@agoric/store';
+import {
+  PaymentPKeywordRecordShape,
+  SeatShape,
+} from '@agoric/zoe/src/typeGuards.js';
+import { AmountShape } from '@agoric/ertp/src/typeGuards.js';
+import { objectMap, deeplyFulfilledObject } from '@agoric/internal';
+
+import { UNPUBLISHED_RESULT } from './offers.js';
+
+/**
+ * @typedef {import('./offers.js').OfferSpec & {
+ *   error?: string,
+ *   numWantsSatisfied?: number
+ *   result?: unknown | typeof import('./offers.js').UNPUBLISHED_RESULT,
+ *   payouts?: AmountKeywordRecord,
+ * }} OfferStatus
+ */
+
+/**
+ * @template {any} T
+ * @typedef {import('@agoric/swingset-liveslots').PromiseWatcher<T, [UserSeat]>} OfferPromiseWatcher<T, [UserSeat]
+ */
+
+/**
+ * @typedef {{
+ * resultWatcher: OfferPromiseWatcher<unknown>,
+ * numWantsWatcher: OfferPromiseWatcher<number>,
+ * paymentWatcher: OfferPromiseWatcher<PaymentPKeywordRecord>,
+ * }} OutcomeWatchers
+ */
+
+/**
+ * @param {OutcomeWatchers} watchers
+ * @param {UserSeat} seat
+ */
+const watchForOfferResult = ({ resultWatcher }, seat) => {
+  const p = E(seat).getOfferResult();
+  watchPromise(p, resultWatcher, seat);
+  return p;
+};
+
+/**
+ * @param {OutcomeWatchers} watchers
+ * @param {UserSeat} seat
+ */
+const watchForNumWants = ({ numWantsWatcher }, seat) => {
+  const p = E(seat).numWantsSatisfied();
+  watchPromise(p, numWantsWatcher, seat);
+  return p;
+};
+
+/**
+ * @param {OutcomeWatchers} watchers
+ * @param {UserSeat} seat
+ */
+const watchForPayout = ({ paymentWatcher }, seat) => {
+  const p = E(seat).getPayouts();
+  watchPromise(p, paymentWatcher, seat);
+  return p;
+};
+
+/**
+ *
+ * @param {OutcomeWatchers} watchers
+ * @param {UserSeat} seat
+ */
+export const watchOfferOutcomes = (watchers, seat) => {
+  return Promise.all([
+    watchForOfferResult(watchers, seat),
+    watchForNumWants(watchers, seat),
+    watchForPayout(watchers, seat),
+  ]);
+};
+
+const offerWatcherGuard = harden({
+  helper: M.interface('InstanceAdminStorage', {
+    updateStatus: M.call(M.any()).returns(),
+    onNewContinuingOffer: M.call(
+      M.or(M.number(), M.string()),
+      AmountShape,
+      M.any(),
+    )
+      .optional(M.record())
+      .returns(),
+    publishResult: M.call(M.any()).returns(),
+    exitOpenSeat: M.call().returns(),
+  }),
+  paymentWatcher: M.interface('paymentWatcher', {
+    onFulfilled: M.call(PaymentPKeywordRecordShape, SeatShape).returns(
+      M.promise(),
+    ),
+    onRejected: M.call(M.any(), SeatShape).returns(),
+  }),
+  resultWatcher: M.interface('resultWatcher', {
+    onFulfilled: M.call(M.any(), SeatShape).returns(),
+    onRejected: M.call(M.any(), SeatShape).returns(),
+  }),
+  numWantsWatcher: M.interface('numWantsWatcher', {
+    onFulfilled: M.call(M.number(), SeatShape).returns(),
+    onRejected: M.call(M.any(), SeatShape).returns(),
+  }),
+});
+
+export const makeOfferWatcherMaker = baggage => {
```

Comment:
this is our convention:
```suggestion
export const prepareOfferWatcher = baggage => {
```

Reply:
fixed